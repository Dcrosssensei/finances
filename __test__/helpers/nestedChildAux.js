export const getNestedChild = (obj, depth) => {
    if (depth === 0 || !obj?.children) return obj;
    return getNestedChild(obj.children[0], depth - 1);
  }
  
  