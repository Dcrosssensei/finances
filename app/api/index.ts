export * from './helpers/transforClass';
export * from './operations';
export * from './validation';
