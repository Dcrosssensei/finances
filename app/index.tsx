import { createStackNavigator } from '@react-navigation/stack';
import Home from "./pages/home/Home";
import Details from './pages/details/Details';
import { RootStackParamList } from './routes/navigationTypes';
import Header from './components/header/Header';
import { RecordProvider } from './context/MainContext';

const Stack = createStackNavigator<RootStackParamList>();


export default function Index() {

  return (
    <RecordProvider>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={Home} options={{ headerShown: false }} />
        <Stack.Screen name="Details" component={Details} 
        options={ {
          header: ()=> (<Header back />)
        }}  
        />
      </Stack.Navigator>
    </RecordProvider>
  );
}
