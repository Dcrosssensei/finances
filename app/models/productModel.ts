export interface ProductTypes {
    id: string;
    name: string;
    description: string;
    logo: string;
    date_release: Date;
    date_revision: Date;
  }

  export interface ProductTypesToSend extends Omit<ProductTypes, 'date_release' | 'date_revision'> {
    date_release: string;
    date_revision: string;
  }
  