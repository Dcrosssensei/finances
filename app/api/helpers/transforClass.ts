import { ProductTypes, ProductTypesToSend } from '@/app/models';
import { format } from 'date-fns';

export function transformToSend(data: ProductTypes): ProductTypesToSend {
  return {
    ...data, 
    date_release: format(data.date_release, 'yyyy-MM-dd'), 
    date_revision: format(data.date_revision, 'yyyy-MM-dd') 
  };
}
