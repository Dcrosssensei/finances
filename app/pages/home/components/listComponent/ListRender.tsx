import { FlatList, ListRenderItemInfo, StyleSheet} from 'react-native'
import React from 'react'
import RenderItemList from './components/RenderItemList'
import { ProductTypes } from '@/app/models'

interface listProp{
    data: ProductTypes[]
    selectItem: (item: ProductTypes, Index: number) => void
}

const ListRender = ({data, selectItem}:listProp) => {
  return (
    <>
      <FlatList
      data={data}
      renderItem={({ item, index }: ListRenderItemInfo<ProductTypes>) => {
          return (
              <RenderItemList item={item} index={index} selectItem={selectItem} />
            )
        }}   
        /> 
    </>
  )
}

export default ListRender

const styles = StyleSheet.create({
    
})