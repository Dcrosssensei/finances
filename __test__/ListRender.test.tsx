import ListRender from "@/app/pages/home/components/listComponent/ListRender";
import React from "react";
import renderer from "react-test-renderer";
import {getNestedChild} from './helpers/nestedChildAux'
import { ProductTypes } from "@/app/models";

describe('RenderItemList', () => {
    let tree: any
    const itemsTest = [{
        "id": "001",
        "name": "Producto Innovador A",
        "description": "Producto A ofrece una solución integral con características avanzadas que mejoran la eficiencia y la productividad.",
        "logo": "https://handsonbanking.org/wp-content/uploads/2021/02/debitcard_front_blue.png",
        "date_release": new Date("2024-01-29"),
        "date_revision": new Date("2024-06-29")
      },
      {
        "id": "product-id-002",
        "name": "Producto Revolucionario B",
        "description": "Producto B es conocido por su durabilidad y rendimiento superior en diversas condiciones operativas.",
        "logo": "https://handsonbanking.org/wp-content/uploads/2021/02/debitcard_front_blue.png",
        "date_release": new Date("2024-01-29"),
        "date_revision": new Date("2024-06-29")
      }
    ]

    beforeAll(()=>{
        tree = renderer.create(
        <ListRender 
            data={itemsTest}
            selectItem={(item, index)=>{}}
        />
        ).toJSON();
    })
    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
      });

    it('renders name first product', () => {
        
        const nameRendered =getNestedChild(tree,6);
        expect(nameRendered).toEqual("Producto Innovador A");
    });
    
    it('renders name second product', () => {
        const secondBranch =tree.children[0].children[1]
        const nameRendered =getNestedChild(secondBranch,4)
        expect(nameRendered).toEqual("Producto Revolucionario B");
    });
    
    it('renders first ID', () => {
        const firstChange =getNestedChild(tree,4);
        const idRendered = firstChange.children[1].children[1]
        expect(idRendered).toEqual("001");
      });

    it('renders second ID', () => {
        const secondBranch =tree.children[0].children[1]
        const firstChange =getNestedChild(secondBranch,2);
        const idRendered = firstChange.children[1].children[1]
        expect(idRendered).toEqual("product-id-002");
      });


  });

describe('RenderItemList without data', () => {
    let tree: any
    const itemsTest: ProductTypes[]  = []

    beforeAll(()=>{
        tree = renderer.create(
        <ListRender 
            data={itemsTest}
            selectItem={(item, index)=>{}}
        />
        ).toJSON();
    })
    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
      });

      it('renders message', () => {
        const nameRendered = getNestedChild(tree,5);
        expect(nameRendered).toEqual("No se encontraron Resultados");
      });

  });
