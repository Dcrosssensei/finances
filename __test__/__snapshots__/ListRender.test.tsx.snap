// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderItemList renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={<ListEmptyComponent />}
  data={
    [
      {
        "date_release": 2024-01-29T00:00:00.000Z,
        "date_revision": 2024-06-29T00:00:00.000Z,
        "description": "Producto A ofrece una solución integral con características avanzadas que mejoran la eficiencia y la productividad.",
        "id": "001",
        "logo": "https://handsonbanking.org/wp-content/uploads/2021/02/debitcard_front_blue.png",
        "name": "Producto Innovador A",
      },
      {
        "date_release": 2024-01-29T00:00:00.000Z,
        "date_revision": 2024-06-29T00:00:00.000Z,
        "description": "Producto B es conocido por su durabilidad y rendimiento superior en diversas condiciones operativas.",
        "id": "product-id-002",
        "logo": "https://handsonbanking.org/wp-content/uploads/2021/02/debitcard_front_blue.png",
        "name": "Producto Revolucionario B",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "lightgray",
            "borderWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 10,
            "paddingVertical": 14,
          }
        }
      >
        <View
          style={
            {
              "flex": 9,
            }
          }
        >
          <Text
            style={
              {
                "fontWeight": "700",
              }
            }
          >
            Producto Innovador A
          </Text>
          <Text
            style={
              {
                "color": "#636262",
                "fontWeight": "500",
              }
            }
          >
            ID: 
            001
          </Text>
        </View>
        <Text />
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "lightgray",
            "borderWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 10,
            "paddingVertical": 14,
          }
        }
      >
        <View
          style={
            {
              "flex": 9,
            }
          }
        >
          <Text
            style={
              {
                "fontWeight": "700",
              }
            }
          >
            Producto Revolucionario B
          </Text>
          <Text
            style={
              {
                "color": "#636262",
                "fontWeight": "500",
              }
            }
          >
            ID: 
            product-id-002
          </Text>
        </View>
        <Text />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`RenderItemList without data renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={<ListEmptyComponent />}
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "height": 60,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "fontSize": 20,
          }
        }
      >
        No se encontraron Resultados
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
